### PR TITLE
Avoid import tensorflow directly

### PR DESCRIPTION
**Pull request by haifeng-jin**
People may use other backends without installing TF.

### CODE REVIEW

File: keras/distribution/distribution_lib.py [hunk 1]
```diff
@@ -14,13 +14,11 @@
...
 import numpy as np
 
-# TF is used for dataset sharding.
-import tensorflow as tf
-
 from keras.api_export import keras_export
 from keras.backend import KerasTensor
 from keras.backend import distribution_lib
 from keras.backend.common import global_state
+from keras.utils.module_utils import tensorflow as tf
```
Comment (qlzh727) [inline keras/distribution/distribution_lib.py:21]: Thanks for the fix. Can we move this to the place that tf is used (eg the distribute_dataset method.)

**Review by qlzh727 — CHANGES_REQUESTED** [review]

**Comment by fchollet** [comment]
Why was this not caught by our dedicated integration test?

**Comment by haifeng-jin** [comment]
> Why was this not caught by our dedicated integration test?

I don't think we uninstalled tensorflow for the jax and pytorch backend tests. Should we? or there are some test cases that always uses tensorflow no matter what backend it is.

**Review by fchollet — APPROVED** [review]
LGTM

**Comment by qlzh727** [comment]
Thanks for the fix.

**Pull request merged**